test/unit/FieldsetDate.test.js

```diff
@@ -0,0 +1,163 @@
+      expect(component.find('fieldset').length).to.equal(1);
+    });
+
+    it('should render a legend with text Date.', () => {
+      expect(component.find('legend').length).to.equal(1);
+      expect(component.find('legend').text()).to.equal('Date');
+    });
+
+    it('should have inputs and labels for start year and end year.', () => {
+      const container = component.find('#input-container');
+
+      expect(container.find('label').length).to.equal(2);
+
+      expect(container.find('label').at(0).text()).to.equal('Start Year');
+      expect(container.find('label').at(0).find('input').length).to.equal(1);
+      expect(container.find('label').at(0).find('input').props().type).to.equal('text');
+      expect(container.find('label').at(0).props().id).to.equal('startDate-label');
+      expect(container.find('label').at(0).find('input').props()['aria-labelledby']).to.equal(
+        'startDate-label dateInput-status'
+      );
+
+      expect(container.find('label').at(1).text()).to.equal('End Year');
+      expect(container.find('label').at(1).find('input').length).to.equal(1);
+      expect(container.find('label').at(1).find('input').props().type).to.equal('text');
+      expect(container.find('label').at(1).props().id).to.equal('endDate-label');
+      expect(container.find('label').at(1).find('input').props()['aria-labelledby']).to.equal(
+        'endDate-label dateInput-status'
+      );
+    });
+
+    it('should have the default state of dateAfter and dateBefore of "0".', () => {
+      expect(component.state('dateAfter')).to.equal('0');
+      expect(component.state('dateBefore')).to.equal('0');
+    });
+
+    it('should have the default state of selectedFacets as the props passed.', () => {
+      expect(component.state('selectedFacets')).to.equal(selectedFacets);
+    });
+
+    it('should render no error messages', () => {
+      expect(component.find('#dateInput-status').text()).to.equal('');
+    });
+  });
+
+  describe('If entering dates', () => {
+    let component;
+    let updateSelectedFacets;
+    const selectedFacets = {
+      language: ['lang:fr', 'lang:sp'],
+    };
+    const calledWithStartDateValues = {
+      language: ['lang:fr', 'lang:sp'],
+      dateAfter: '2001',
+    };
+    const calledWithEndDateValues = {
+      language: ['lang:fr', 'lang:sp'],
+      dateAfter: '2001',
+      dateBefore: '2100',
+    };
+
+    before(() => {
+      component = mount(<FieldsetDate selectedFacets={selectedFacets} />);
+      updateSelectedFacets = sinon.spy(Actions, 'updateSelectedFacets');
+    });
+
+    afterEach(() => {
+      updateSelectedFacets.reset();
+    });
+
+    after(() => {
+      updateSelectedFacets.restore();
+      component.unmount();
+    });
+
+    it('should update selectedFacets based on it\'s input from Start Year input.', () => {
+      const startYearInput = component.find('#input-container').find('label').at(0).find('input');
+      const getCustomEvent = (value, name) => {
```

Comment:
This is the same function as what as in the `react-number-format` module. 👍 

You have the same function TWICE in two different `it` statements, so you should move it outside the tests so you only declare it once. Also write a comment about it and how it's from the `react-number-format` module and used for testing.

Reply:
Got it! Will do.